Reduce memory when merging partial fixture files

I was looking at the pypy memory issue today and (at least in very recent runs) observed that the memory seems to be triggered during `pytest_sessionfinish()` in `merge_partial_fixture_files()`:

```text
    File "/opt/actions-runner/_work/execution-specs/execution-specs/packages/testing/src/execution_testing/cli/pytest_commands/     
  plugins/filler/filler.py", line 1653, in pytest_sessionfinish                                                                     
      merge_partial_fixture_files(fixture_output.directory)                                                                         
    File                                                                                                                            
  "/opt/actions-runner/_work/execution-specs/execution-specs/packages/testing/src/execution_testing/fixtures/collector.py",         
  line 83, in merge_partial_fixture_files                                                                                           
      target_path.write_text("".join(parts))                                                                                        
    File "/opt/actions-runner/_work/_tool/PyPy/3.11.13/x64/lib/pypy3.11/pathlib.py", line 1078, in write_text                       
      with self.open(mode='w', encoding=encoding, errors=errors, newline=newline) as f:                                             
  RuntimeError: reentrant call                                                                                                      
  ......................................................            [20399/20399]pypy3: exit 1 (861.25 seconds)                     
  /opt/actions-runner/_work/execution-specs/execution-specs> fill --skip-index --no-html --tb=no --show-capture=no                  
  --disable-warnings -m 'not slow and not benchmark and not derived_test' -n auto --maxprocesses 7 --dist=loadgroup                 
  --basetemp=/opt/actions-runner/_work/execution-specs/execution-specs/.tox/.tmp/pytest --log-to                                    
  /opt/actions-runner/_work/execution-specs/execution-specs/.tox/logs --clean --fork Amsterdam tests pid=5845                       
    pypy3: FAIL code 1 (908.98=setup[47.72]+cmd[861.25] seconds)                                                                    
    evaluation failed :( (909.15 seconds)                                                                                           
  Error: Process completed with exit code 1.  
```

This traceback is from: https://github.com/ethereum/execution-specs/actions/runs/21623955209/job/62319488584?pr=2116#step:5:433 

The PR that triggered this, #2116, tried to add a hotfix to reduce memory by avoiding filling all formats and only filling Amsterdam.

One possible solution is to avoid reading all the partial fixtures into memory here and "stream" them instead (i.e., read and write each JSON line-by-line, instead of all JSON):
https://github.com/ethereum/execution-specs/blob/0c65695a03c592e73dec3521f971564d5ff6caf5/packages/testing/src/execution_testing/fixtures/collector.py#L63-L83

--- a/packages/testing/src/execution_testing/fixtures/collector.py
+++ b/packages/testing/src/execution_testing/fixtures/collector.py
@@ -71,16 +71,15 @@ def merge_partial_fixture_files(output_dir: Path) -> None:
 
         # Write final JSON file
         sorted_keys = sorted(entries.keys())
-        parts = ["{\n"]
         last_idx = len(sorted_keys) - 1
-        for i, key in enumerate(sorted_keys):
-            key_json = json.dumps(key)
-            # Add indentation for nesting inside outer JSON object
-            value_indented = entries[key].replace("\n", "\n    ")
-            parts.append(f"    {key_json}: {value_indented}")
-            parts.append(",\n" if i < last_idx else "\n")
-        parts.append("}")
-        target_path.write_text("".join(parts))
+        with open(target_path, "w") as f:
+            f.write("{\n")
+            for i, key in enumerate(sorted_keys):
+                key_json = json.dumps(key)
+                value_indented = entries[key].replace("\n", "\n    ")
+                f.write(f"    {key_json}: {value_indented}")
+                f.write(",\n" if i < last_idx else "\n")
+            f.write("}")
 
         # Clean up partial files
         for partial in partials:

